diff --git env-hooks/99.roslisp.sh env-hooks/99.roslisp.sh
index a2733d8..62c5931 100644
--- env-hooks/99.roslisp.sh
+++ env-hooks/99.roslisp.sh
@@ -18,4 +18,4 @@ for workspace in workspaces:
 print(os.pathsep.join(paths))
 EOF
 )
-export ROSLISP_PACKAGE_DIRECTORIES="`python -c \"$PYTHON_CODE_BUILD_ROSLISP_PACKAGE_DIRECTORIES\"`"
+export ROSLISP_PACKAGE_DIRECTORIES="`python$ROS_PYTHON_VERSION -c \"$PYTHON_CODE_BUILD_ROSLISP_PACKAGE_DIRECTORIES\"`"
